### ISSUE TITLE
[Runtime Env] Runtime Environments eager install;  does not work

### ISSUE THREAD

**Issue by frosk1** [ray-project/ray#23767]
### What happened + What you expected to happen

1. When using runtime_env  and the attribute eager_install = False on a task twice, the second time still runs the installation again instead of using the already installed environment.

2. The second run with equal runtime_env should use the already installed runtime_env instead of creating a new one and installing all dependencies again.

### Versions / Dependencies

Ubuntu 18.04.6 LTS
ray, version 1.11.0
Python 3.7.5

### Reproduction script

script.py
```
import ray

@ray.remote(runtime_env={"pip": ["lightgbm"], "eager_install": False})
def get_version():
    import lightgbm as lgb
    print(f"lightgmb version: {lgb.__version__}")

if __name__ == '__main__':
    ray.init(address="auto")
    ray.get(get_version.remote())
```

On local machine:

1. Run: `ray start --head`
2. Run:  `python3  script.py`   ---->  check that dependencies get installed
3. Run:  `python3  script.py`   ---->  check that dependencies get installed again, but they should not.

**Comment by frosk1** [comment]
https://github.com/ray-project/ray/tree/1.12.0

**Comment by architkulkarni** [comment]
That's great that this is fixed in 1.12.0!  Feel free to reopen if the issue persists.
